### DISCUSSION TITLE
Upgrade to Testing changed my Spiral theme.

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#240] [Q&A]
I upgraded last night from Bullseye to Bookworm.  I followed the instructions on the SpiralLinux Wiki.  But after I rebooted, my desktop was changed to whatever the Debian maintainers wanted for LXQt.  I can probably rollback the system with Snapper, but how do I ensure the changes from Stable to Testing don't nuke all of Spiral's beautiful settings?

Is there a program or script I can use to preserve my Spiral settings and themes?  Really excited about Bookworm.

I realize this is most likely not a problem with SpiralLinux itself, but user/pilot error on my part.  Do I just need to get familiar with the Debian docs?  What's going on here?

**Comment by geckolinux** [comment]
Hi there, could you please post some screenshots with the different theme configuration modules visible? I'll have to test the different desktop environments after upgrading to Bookworm, but in general LXQt is the hardest to theme and the least stable in terms of theming, so it's not your fault and it will probably require a bit of manual fixing.

**Reply by birds-swim** [reply]
> in general LXQt is the hardest to theme and the least stable in terms of theming

Well that sucks.  =/  For the both of us, tbh.  Yeah, I can give you screenshots.  I think obvious screenshots would be of the desktop and start menu.  But what are the "modules" you're talking about?  Is that a file?  Are you looking for the output of a file?  Or is this strictly a visual thing?  What's an LXQt module?

I'm willing to help.  Help me help you?

**Comment by zparihar** [comment]
Hi @birds-swim : You said there was Instructions on the Wiki to upgrade from Bullseye to Bookworm.  Would you be able to post a link? I can't seem to find it.  

I'm running Spiral Linux KDE - I think its the best Distro out there at the moment (been running Kubuntu previously for 8 years, but 22.04 did a sub-par good job unfortunately).  Anyways, i've spun up a virtual machine with Spiral KDE and i'm about to change all the sources.list and sources.list.d/*.list from bullseye to bookworm and do an apt update, apt upgrade and apt autoremove

@geckolinux would you say that is the suggested way? 

Thanks!

**Reply by birds-swim** [reply]
https://github.com/SpiralLinux/SpiralLinux-project/wiki

https://wiki.debian.org/DebianTesting

Be advised: Debian has changed non-free to non-free firmware. Please be sure to make the appropriate changes in your apt sources list hen you upgrade to Bookworm. Also change everything from bullseye to bookworm. Do not rely on the words "stable" in your apt sources list. The Debian Wiki says nasty surprises can happen.

**Comment by zparihar** [comment]
I did the following:

sudo sed -i 's/bullseye/bookworm/g' /etc/apt/sources.list
sudo sed -i 's/bullseye/bookworm/g' /etc/apt/sources.list.d/spirallinux.list 
sudo sed -i 's/non-free/non-free non-free-firmware/g' /etc/apt/sources.list
sudo sed -i 's/non-free/non-free non-free-firmware/g' /etc/apt/sources.list.d/spirallinux.list
sudo apt update
sudo apt upgrade -y
sudo apt autoremove
sudo shutdown -r now

This upgrade many things, but didn't upgrade KDE... it was still at 5.20

So then I ran:

sudo apt install plasma-browser-integration plasma-dataengines-addons:amd64 plasma-desktop plasma-desktop-data plasma-discover plasma-discover-backend-flatpak plasma-discover-backend-fwupd plasma-discover-backend-snap plasma-discover-common plasma-disks plasma-framework plasma-integration plasma-nm plasma-pa plasma-runner-installer plasma-runners-addons:amd64 plasma-thunderbolt plasma-widgets-addons plasma-workspace plasma-workspace-data kdeconnect kde-telepathy-send-file qgnomeplatform-qt5 qt5-image-formats-plugins qt5-qmltooling-plugins libwacom-bin network-manager-fortisslvpn network-manager-iodine network-manager-l2tp network-manager-openconnect network-manager-openvpn network-manager-pptp network-manager-ssh network-manager-strongswan network-manager-vpnc quota libqt5multimedia5-plugins qml-module-qtquick-virtualkeyboard keditbookmarks kpackagelauncherqml libkf5guiaddons-bin libkf5iconthemes-bin kde-telepathy libkf5iconthemes-bin libkf5parts-plugins qml-module-org-kde-runnermodel kde-telepathy-kaccounts kde-telepathy-kpeople telepathy-mission-control-5 geoclue-2.0 qtspeech5-speechd-plugin libstrongswan-standard-plugins easy-rsa bluedevil fonts-noto ibus-data kde-config-flatpak kde-config-gtk-style kde-config-sddm kde-style-oxygen-qt5 kgamma5 khelpcenter khotkeys kinfocenter kmenuedit kscreen ksshaskpass kwrited plasma-firewall plasma-systemmonitor plasma-vault plasma-welcome powerdevil xdg-desktop-portal-gtk qml-module-org-kde-activities apt-config-icons-hidpi apt-config-icons-large apt-config-icons-large-hidpi kde-config-updates software-properties-qt kup-backup partitionmanager fonts-noto-hinted ksystemstats qml-module-org-kde-pipewire fonts-noto-cjk fonts-noto-cjk-extra fonts-noto-color-emoji fonts-noto-extra fonts-noto-mono fonts-noto-ui-extra fonts-noto-unhinted kde-config-flatpak plasma-firewall plasma-welcome
sudo shutdown -r now

This now had KDE 5.27.2, but systems settings wasn't looking too great, so I ran:

sudo apt install plasma-marble plasma-wallpapers-addons plasma-calendar-addons plasma-dialer plasma-phonebook plasma-remotecontrollers plasma-runners-addons plasma-settings plasma-widget-foreigncurrencies plasma-workspace-wallpapers libkwinxrenderutils12 kaccounts-integration kaccounts-providers kde-telepathy-kaccounts libkaccounts2 qml-module-org-kde-kaccounts

Systems settings was still not looking too great, so I ran:

sudo apt full-upgrade
sudo shutdown -r now

Now everything looks Great!

I'm just gonna have to test everything out

**Reply by zparihar** [reply]
Plasma Remote Controller is crashing when I'm logging in.  So I removed it with:

sudo apt remove plasma-remotecontrollers
sudo apt autoremove

**Reply by birds-swim** [reply]
Thank you for doing this! I love KDE. It is my favorite. I was using LXQT for performance benefits because I planned on using a lot of security features from my desktop and I expected a performance hit. To compensate, I chose Lxqt.

**Reply by zparihar** [reply]
My pleasure.

My next goal now is to get Spiral Linux working on a ZFS Root File System.
I think i'll need to get some advice from @geckolinux 

Wish me luck!
